client/local_test.go

```diff
@@ -97,18 +98,27 @@ func (s *localLiveSuite) SetUpSuite(c *gc.C) {
 		legacy.SetManagementURL("http://management.test.invalid/url")
 		s.service = legacy
 	}
+	for _, msg := range logMsg {
+		c.Logf(msg)
+	}
+	if s.authMode != identity.AuthLegacy {
+		s.service.SetupHTTP(nil)
+	}
 	s.LiveTests.SetUpSuite(c)
 }
 
 func (s *localLiveSuite) TearDownSuite(c *gc.C) {
 	s.LiveTests.TearDownSuite(c)
 	s.HTTPSuite.TearDownSuite(c)
+	s.service.Stop()
```

Comment:
The service is only started for authMode = AuthLegacy, so stopping it unconditionally doesn't make sense?
It seems that now that there's differences between legacy and non legacy auth, one option is to have a  2 suites, one for legacy auth and one for non-legacy auth. These suites would embed the localLiveSuite suite. The s.service would be moved to legacyLocalLiveSuite and TestAuthenticate would be moved off localLiveSuite and separate implementationd added to each new suite.